src/Halogen/Query/EventSource.purs

```diff
@@ -0,0 +1,71 @@
+module Halogen.Query.EventSource
+  ( EventSourceE()
+  , EventSource()
+  , makeEventSource
+  , runEventSource
+  , interpretEventSource
+  , catEventSource
+  , hoistEventSource
+  , contrahoistEventSource
+  ) where
+
+import Prelude
+
+import Control.Coroutine.Stalling as SCR
+import Control.Monad.Aff (Aff())
+import Control.Monad.Aff.Class (MonadAff)
+import Control.Monad.Free.Trans (hoistFreeT, interpret)
+
+import Data.Bifunctor (lmap)
+import Data.ExistsRE (ExistsRE(), mkExistsRE, runExistsRE)
+import Data.NaturalTransformation (Natural())
+import Data.Functor.Coproduct (Coproduct(), coproduct)
+import Data.Const (Const())
+import Data.Maybe (Maybe(..))
+
+import Unsafe.Coerce as U
+
+newtype EventSourceE f eff = EventSourceE (SCR.StallingProducer (f Unit) (Aff eff) Unit)
+newtype EventSource f (g :: * -> *) = EventSource (ExistsRE (EventSourceE f))
+
+makeEventSource :: forall f g eff. (MonadAff eff g) => SCR.StallingProducer (f Unit) (Aff eff) Unit -> EventSource f g
+makeEventSource = EventSource <<< mkExistsRE <<< EventSourceE
+
+runEventSource :: forall f g eff. (MonadAff eff g) => EventSource f g -> SCR.StallingProducer (f Unit) (Aff eff) Unit
+runEventSource (EventSource xese) = runExistsRE (\(EventSourceE p) -> hoistFreeT coerceEffects p) xese
+  where
+  -- This coercion is safe because our `MonadAff` constraint implies `fff ~ eff`
```

Comment:
:boom::gun: it isn't actually:

Say we `makeEventSource` with `MonadAff (x, y, z) g` I guess a `MonadAff () g` still works for `runEventSource`.